src/hooks.ts

```diff
@@ -0,0 +1,143 @@
+import shallowEqual from "./utils/shallow-equal";
+
+export interface ResultProps {
+  [propName: string]: any;
+}
+
+export function useQueryResult<Data = any, Props = ResultProps>(
+  queryId: string,
+  resultToProps?: (state?: Data) => Props
+): Data | Props | undefined {
+  const { store } = React.useContext(MoonContext);
+  //@ts-ignore can't be undefined
+  const queryResult = store.readQuery<Data>(queryId).data;
+  const [state, setState] = React.useState<Data | undefined>(queryResult);
+  //@ts-ignore can't be undefined
+  store?.subscribeToQueryResult(queryId, setState);
+
+  React.useEffect(() => {
+    //@ts-ignore can't be undefined
+    return () => store?.unsubscribeFromQueryResult(queryId, setState);
+  }, [queryId]);
+
+  return resultToProps ? resultToProps(state) : state;
+}
+
+export function useQueriesResults<Data = any, Props = ResultProps>(
+  queriesIds: string[],
+  resultsToProps?: (results: QueriesResults<Data>) => Props
+): QueriesResults<Data> | Props {
+  const { store } = React.useContext(MoonContext);
+  const { value: currentQueriesIds } = usePrevValue(queriesIds);
+  const queriesStates = currentQueriesIds.reduce<QueriesResults<Data>>((result, queryId) => {
+    //@ts-ignore can't be undefined
+    result[queryId] = store.readQuery<Data>(queryId).data;
+    return result;
+  }, {});
+  const [states, setStates] = React.useState<QueriesResults<Data>>(queriesStates);
+  const setState = (queryId: string) => (result: Data) => {
+    setStates({ ...states, [queryId]: result });
+  };
+
+  const queriesHandlers = React.useMemo(
+    () =>
+      currentQueriesIds.reduce((handlers, queryId) => {
+        handlers[queryId] = setState(queryId);
+        return handlers;
+      }, {}),
+    [currentQueriesIds]
+  );
+  const { prevValue: prevQueriesHandlers } = usePrevValue(queriesHandlers);
+
+  React.useEffect(() => {
+    currentQueriesIds.forEach(queryId => {
+      //@ts-ignore can't be undefined
+      store?.subscribeToQueryResult(queryId, queriesHandlers[queryId]);
+    });
+    return () =>
+      queriesIds.forEach(queryId => {
+        //@ts-ignore can't be undefined
+        store?.unsubscribeFromQueryResult(queryId, prevQueriesHandlers[queryId]);
+      });
+  }, [currentQueriesIds]);
+
+  return resultsToProps ? resultsToProps(states) : states;
+}
+
+export function useQueryState<Data = any, Props = ResultProps>(
+  queryId: string,
+  stateToProps?: (state: QueryState<Data>) => Props
+): QueryState<Data> | Props {
+  const { store } = React.useContext(MoonContext);
+  //@ts-ignore can't be undefined
+  const queryState = store.readQuery(queryId);
+  const [state, setState] = React.useState<QueryState<Data>>(queryState);
+  //@ts-ignore can't be undefined
+  store.subscribeToQuery(queryId, setState);
+
+  React.useEffect(() => {
+    //@ts-ignore can't be undefined
+    return () => store.unsubscribeFromQuery(queryId, setState);
+  }, [queryId]);
```

Comment:
On peut pas simplifier ?
```
//@ts-ignore can't be undefined
React.useEffect(() =>  store.unsubscribeFromQuery(queryId, setState), [queryId]);
```

Reply:
nop ()=>store.unsubscribeFromQuery(queryId, setState) is executed "on unmount"